src/crypto-providers/trezorCryptoProvider.ts

```diff
@@ -362,17 +381,55 @@ const TrezorCryptoProvider: () => Promise<CryptoProvider> = async () => {
     })
   }
 
-  const signTx = async (
+  const createWitnesses = (
+    trezorWitnesses: TrezorTypes.CardanoSignedTxWitness[],
+    signingFiles: HwSigningData[],
+  ): TxWitnesses => {
+    const getSigningFileDataByXPubKey = (pubKey: PubKeyHex): HwSigningData => {
+      const hwSigningData = signingFiles.find(
+        (signingFile) => splitXPubKeyCborHex(signingFile.cborXPubKeyHex).pubKey.toString('hex') === pubKey,
+      )
+      if (hwSigningData) return hwSigningData
+      throw Error(Errors.MissingHwSigningDataAtXPubKeyError)
+    }
+
+    const [byronWitnesses, shelleyWitnesses] = partition(
+      trezorWitnesses.map((witness) => {
+        const signingFile = getSigningFileDataByXPubKey(witness.pubKey as PubKeyHex)
+        return ({
+          ...witness,
```

Comment:
I would prefer if we were explicit about what goes into the return value. The type `TrezorTypes.CardanoSignedTxWitness` is not even part of this repository (and my VS code fails to navigate it).

The same for Ledger, I guess.